src/gui/main_gui.cpp

```diff
@@ -33,8 +33,17 @@ GUITest(World world) {
 
     auto mesh_trees = entity::generate_mesh(default_trees_voxel);
 
-    //glEnable( GL_DEBUG_OUTPUT );
-    //glDebugMessageCallback( MessageCallback, 0 );
+    glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, true);
+    int context_flag;
+    glGetIntegerv(GL_CONTEXT_FLAGS, &context_flag);
+    if (context_flag & GL_CONTEXT_FLAG_DEBUG_BIT) {
+        glEnable(GL_DEBUG_OUTPUT);
+        glEnable(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+        glDebugMessageCallback(message_callback, 0);
+        glDebugMessageControl(
+            GL_DONT_CARE, GL_DONT_CARE, GL_DONT_CARE, 0, nullptr, GL_TRUE
+        );
+    }
```

Comment:
Add a comment explaining what this does.

Also use a `uintXX_t` for `context_flag`.